Anytype/Sources/PresentationLayer/Common/SwiftUI/SyncStatus/SyncStatusInfoViewModel.swift

```diff
@@ -5,13 +5,17 @@ import Services
 final class SyncStatusInfoViewModel: ObservableObject {
     @Injected(\.syncStatusStorage)
     private var syncStatusStorage: SyncStatusStorageProtocol
+    @Injected(\.p2pStatusStorage)
+    private var p2pStatusStorage: P2PStatusStorageProtocol
     
     @Published var syncStatusInfo: SyncStatusInfo
+    @Published var p2pStatusInfo: P2PStatusInfo
     
     init(spaceId: String) {
         syncStatusInfo = .default(spaceId: spaceId)
+        p2pStatusInfo = .default(spaceId: spaceId)
         
         Task { await syncStatusStorage.statusPublisher(spaceId: spaceId).assign(to: &$syncStatusInfo) }
-    }
-    
+        Task { await p2pStatusStorage.statusPublisher(spaceId: spaceId).assign(to: &$p2pStatusInfo) }
```

Comment:
"assing(to:)" called not in main actor. I think make async method that receive publisher is not filexible idea, that can be make a memory problem.

Reply:
Problem 1 https://github.com/anyproto/anytype-swift/pull/1780/files#r1676057091
Problem 2 https://github.com/anyproto/anytype-swift/pull/1780/files#r1676047881

Reply:
I will address this in upcoming pr